docs/en/stack/ml/df-analytics/flightdata-regression.asciidoc

```diff
@@ -0,0 +1,367 @@
+        },
+        {
+          "phase" : "writing_results",
+          "progress_percent" : 100
+        }
+      ]
+    }
+  ]
+}
+----  
+
+
+The job has four phases. When all the phases have completed, the job 
+state becomes `stopped` and the results are ready to view and evaluate.
+--
+
+
+[[flightdata-regression-results]]
+==== Viewing results
+
+. Run the following command to view the results in the destination index:
++
+--
+
+[source,console]
+--------------------------------------------------
+GET df-flight-delays/_search
+--------------------------------------------------
+// TEST[skip:TBD]
+
+
+The snippet below shows a part of a document with the annotated results:
+
+[source,console-result]
+----  
+          ...
+          "DestRegion" : "UK",
+          "OriginAirportID" : "LHR",
+          "DestCityName" : "London",
+          "FlightDelayMin" : 66,      <1>
+          "ml" : {
+            "FlightDelayMin_prediction" : 62.527,   <2>
+            "is_training" : false   <3>
+          }
+          ...
+----
+
+<1> The `dependent_variable` with the ground truth value. This is what we are 
+trying to predict with the {reganalysis}.
+<2> The prediction. The field name is suffixed with `_prediction`.
+<3> Indicates that this document was not used in the training set.
+
+
+If a document doesn't contain a prediction field, then it is excluded from the 
+analysis. In order to be analyzed, a document must contain at least one field 
+with a supported data type (i.e. `numeric`, `boolean`, `text`, `keyword` or 
+`ip`) and must not contain arrays with more than one item.
+--
+
+
+[[flightdata-regression-evaluate]]
+==== Evaluating results
+
+The results can be evaluated for documents which contain both the ground truth 
+field and the prediction. In the example below, `FlightDelayMins` contains the 
+ground truth and the prediction is stored as `ml.FlightDelayMin_prediction`.
+
+. Use the {dfanalytics} evaluate API to evaluate the results.
++
+--
+First, we want to know the training error that represents how well the model 
+performed on the training dataset:
+
+[source,console]
+--------------------------------------------------
+POST _ml/data_frame/_evaluate
+{
+ "index": "df-flight-delays",   <1>
+  "query": {
+      "bool": {
+        "filter": [{ "term":  { "ml.is_training": true } }]  <2>
+      }
+    },
+ "evaluation": {
+   "regression": {
+     "actual_field": "FlightDelayMin",   <3>
+     "predicted_field": "ml.FlightDelayMin_prediction", <4>
+     "metrics": {  
+       "r_squared": {},
+       "mean_squared_error": {}                            
+     }
+   }
+ }
+}
+--------------------------------------------------
+// TEST[skip:TBD]
+
+<1> The destination index which is the output of the analysis job.
+<2> We calculate the training error by only evaluating the training data.
+<3> The ground truth label.
+<4> Predicted value.
+
+
+You can also evaluate the data that has not been used to train the model, the 
```

Comment:
We should give similar emphasis on evaluating training and generalization errors. E.g. something along the lines of...

`Next, we calculate the generalization error that represents how well the model performed on previously unseen data.`